### PR TITLE
SW-2667 Add a user-provider in the web-app react tree

### PR DESCRIPTION
**Pull request by karthikbtf**
- this PR primarily introduces how to use a provider
- introduced a directory and export structure, this is totally changeable and not set in stone
- application of user provider is a bit forced at this point since we don't have that many uses
- we can re-evaluate this later, good start for now
- tested all occurrences of user/reloadUser
- cleaned up props

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -750,6 +742,13 @@ export default function App() {
           </ErrorBoundary>
         </div>
       </div>
+    </>
+  );
+
+  return (
+    <StyledEngineProvider injectFirst>
+      <CssBaseline />
+      <UserProvider data={{ user, reloadUser }}>{getContent()}</UserProvider>
```
Comment (karthikbtf) [inline src/App.tsx:751]: ideally we fetch the user within the provider layer, for now i'm keeping the existing fetching sequence just not to break anything. will revisit this later.

File: src/components/NavBar.tsx [hunk 2]
```diff
@@ -94,6 +94,7 @@ export default function NavBar({
           closeAndNavigateTo(APP_PATHS.INVENTORY);
         }}
         id='inventory'
+        key='inventory'
```
Comment (constanzauanini) [inline src/components/NavBar.tsx:97]: What is this key for?

**Review by constanzauanini — APPROVED** [review]
Looks good!

File: src/components/NavBar.tsx (hunk 2 above)
Comment (karthikbtf) [reply src/components/NavBar.tsx:97]: This is an array of nav items, react was complaining the elements needed keys.

Comment (constanzauanini) [reply src/components/NavBar.tsx:97]: Thanks!

File: src/components/MyAccount/index.tsx [hunk 3]
```diff
@@ -33,14 +34,36 @@ const columns: TableColumnType[] = [
...
+type MyAccountContentProps = {
   user: User;
   organizations?: ServerOrganization[];
   edit: boolean;
   reloadUser: () => void;
   reloadData?: () => void;
 };
 
-export default function MyAccount({ user, organizations, edit, reloadUser, reloadData }: MyAccountProps): JSX.Element {
+const MyAccountContent = ({
```
Comment (charlie-tf) [inline src/components/MyAccount/index.tsx:60]: I like your refactor strategy here 👍

**Review by charlie-tf — APPROVED** [review]
Nice!

File: src/providers/hooks.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { useContext } from 'react';
+import { UserContext } from './contexts';
+
+export const useUser = () => useContext(UserContext);
```
[the remark this answers is not in the document]
Comment (charlie-tf) [reply src/providers/hooks.ts:4]: I like it!

**Pull request merged**